lms/djangoapps/program_enrollments/api/api.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+"""
+ProgramEnrollment internal api
+"""
+from __future__ import absolute_import, unicode_literals
+
+from datetime import datetime, timedelta
+from pytz import UTC
+
+from django.urls import reverse
+
+from six import iteritems
+
+from bulk_email.api import is_bulk_email_feature_enabled, is_user_opted_out_for_course
+from edx_when.api import get_dates_for_course
+from xmodule.modulestore.django import modulestore
+from lms.djangoapps.program_enrollments.api.v1.constants import (
+    CourseEnrollmentResponseStatuses,
+    CourseRunProgressStatuses,
+    MAX_ENROLLMENT_RECORDS,
+    ProgramEnrollmentResponseStatuses,
+)
+
+
+def get_due_dates(request, course_key, user):
+    """
+    Get due date information for a user for blocks in a course.
+
+    Arguments:
+        request: the request object
+        course_key (CourseKey): the CourseKey for the course
+        user: the user object for which we want due date information
+
+    Returns:
+        due_dates (list): a list of dictionaries containing due date information
+            keys:
+                name: the display name of the block
+                url: the deep link to the block
+                date: the due date for the block
+    """
+    dates = get_dates_for_course(
+        course_key,
+        user,
+    )
+
+    store = modulestore()
+
+    due_dates = []
+    for (block_key, date_type), date in iteritems(dates):
+        if date_type == 'due':
+            block = store.get_item(block_key)
+
+            # get url to the block in the course
+            block_url = reverse('jump_to', args=[course_key, block_key])
+            block_url = request.build_absolute_uri(block_url)
```

Comment:
Does this URL actually get used by the frontend? Seems like we will have to rewrite the URL in order to make it work within the context of the learner portal. Wondering if it makes sense to include it here at all.

Reply:
Good question, I was told the FE would need it, so I've included it.

Reply:
i think it is good to leave it for now, as the intended behavior is for the user to click on this and be taken to a specific part of the course. when we actually do the work to have the courseware pulled in as a microfrontend, we can update the return value appropriately.